vnext/Shared/DevSupportManager.cpp

```diff
@@ -246,6 +246,7 @@ void DevSupportManager::EnsureHermesInspector(
 #ifdef HERMES_ENABLE_DEBUGGER
   static std::once_flag once;
   std::call_once(once, [this, &packagerHost, packagerPort]() {
+    // TODO: should we use the bundleAppId as the app param if available?
```

Comment:
It would probably be good to work out how to plug this through.  The inspector proxy is the only usage of the appId I can find in the OSS code.  [https://github.com/facebook/metro/blob/a0ba9945ea0bcb99e2048eb76dafafa03a0e1709/packages/metro-inspector-proxy/src/InspectorProxy.js#L167]

I'm guessing Meta has an extra front end on the metro server that makes use of this parameter?  Right now its use feels fairly limited for OSS users.

Reply:
I think it's tailored to unblock per app metro configuration for users that want to serve multiple app bundles from a single metro instance in a monorepo. So in that sense, yes, it is a critical feature for Meta, but I could imagine it could be useful even on a smaller scale with a handful of apps in a monorepo.

Reply:
Another follow up question is whether we should default to sending the Windows AppPackage identifier when BundleAppId is unspecified in ReactInstanceSettings

Reply:
For extra context:
1. RN passes the `app` parameter to bundle requests and to `/inspector/device` requests, but the two parameters aren't coupled on the Metro server side. (The inspector proxy is its own server that is merely started by Metro.)
2. In the inspector proxy, it looks like the app ID is used for logging and possibly for display purposes in the debugger frontend (via the "page description" object)
3. In Metro, the `app` parameter in bundle requests is visible to [`rewriteRequestUrl`](https://facebook.github.io/metro/docs/configuration/#rewriterequesturl), which is the extension point we use at Meta to customise bundle options per app in our monorepo.

Reply:
Thanks for the info @motiz88.  Just checking - this doesn't allow say different serializers to be used for different apps, right? We have some cases today where it would be nice to be able to run a single metro instance, but a bunch of metro config settings have to be different.  Looks like this would only allow differences that you can control through the url.

@rozele, I wonder if we should include a link or something to `rewriteRequestUrl` in the docs as an example. 
As for defaulting to the package identifier, not all apps are packaged apps, so its not something we can rely on.  And it seems like for most users this parameter is not going to be used, so maybe just keep it simple and leave it off if it's not specified as you've done.

Reply:
> this doesn't allow say different serializers to be used for different apps

Not that I'm aware, but letting the serializer read the `app` param is logically sound and I'd be more than happy to support it in Metro core if it helps your use case.